app/Libraries/OsuAuthorize.php

```diff
@@ -722,7 +722,7 @@ public function checkBeatmapsetDescriptionEdit(?User $user, Beatmapset $beatmaps
     {
         $this->ensureLoggedIn($user);
 
-        if ($user->user_id === $beatmapset->user_id || $user->isModerator()) {
+        if ((!$beatmapset->download_disabled && $user->user_id === $beatmapset->user_id) || $user->isModerator()) {
```

Comment:
does this handle the case where `download_disabled` is `false` but `download_disabled_url` is non-null? if not, it should be (along with any similar usages). rationale is that most takedown requests these days are focused on specific files, not the whole beatmap content, and we leave the pieces that are still okay to host.